src/compiler/types.ts

```diff
@@ -920,6 +920,14 @@ export const enum RelationComparisonResult {
     ReportsMask         = ReportsUnmeasurable | ReportsUnreliable,
 }
 
+/** @internal */
+export const enum PredicateSemantics {

```

Comment:
I think `Ternary` would actually cover this, if you didn't want to come up with something new.